datumaro/cli/commands/download.py

```diff
@@ -0,0 +1,125 @@
+        """.format(available_datasets, ', '.join(builtin_writers)),
+        formatter_class=MultilineFormatter)
+
+    parser.add_argument('-i', '--dataset-id', required=True,
+        help="Which dataset to download")
+    parser.add_argument('-f', '--output-format',
+        help="Output format (default: original format of the dataset)")
+    parser.add_argument('-o', '--output-dir', dest='dst_dir',
+        help="Directory to save output (default: a subdir in the current one)")
+    parser.add_argument('--overwrite', action='store_true',
+        help="Overwrite existing files in the save directory")
+    parser.add_argument('extra_args', nargs=argparse.REMAINDER,
+        help="Additional arguments for output format (pass '-- -h' for help). "
+            "Must be specified after the main command arguments")
+    parser.set_defaults(command=download_command)
+
+    return parser
+
+def get_sensitive_args():
+    return {
+        download_command: ['dst_dir', 'extra_args'],
+    }
+
+def download_command(args):
+    env = Environment()
+
+    if args.dataset_id.startswith('tfds:'):
+        if TFDS_EXTRACTOR_AVAILABLE:
+            tfds_ds_name = args.dataset_id[5:]
+            tfds_ds_metadata = AVAILABLE_TFDS_DATASETS.get(tfds_ds_name)
+            if tfds_ds_metadata:
+                default_converter_name = tfds_ds_metadata.default_converter_name
+                extractor_factory = lambda: make_tfds_extractor(tfds_ds_name)
+            else:
+                raise CliException(f"Unsupported TFDS dataset '{tfds_ds_name}'")
+        else:
+            raise CliException(
+                "TFDS datasets are not available, because TFDS and/or "
+                    "TensorFlow are not installed.\n"
+                "You can install them with: pip install datumaro[tf,tfds]")
+    else:
+        raise CliException(f"Unknown dataset ID '{args.dataset_id}'")
+
+    try:
+        converter = env.converters[args.output_format or default_converter_name]
+    except KeyError:
+        raise CliException("Converter for format '%s' is not found" %
+            args.output_format)
+    extra_args = converter.parse_cmdline(args.extra_args)
+
+    dst_dir = args.dst_dir
+    if dst_dir:
+        if not args.overwrite and osp.isdir(dst_dir) and os.listdir(dst_dir):
+            raise CliException("Directory '%s' already exists "
+                "(pass --overwrite to overwrite)" % dst_dir)
+    else:
+        dst_dir = generate_next_file_name('%s-%s' % (
+            make_file_name(args.dataset_id),
+            make_file_name(args.output_format),
+        ))
+    dst_dir = osp.abspath(dst_dir)
+
+    log.info("Downloading the dataset")
+    extractor = extractor_factory()
+
+    log.info("Exporting the dataset")
+    converter.convert(extractor, dst_dir,
+        default_image_ext='.png', **extra_args)
+
+    log.info("Dataset exported to '%s' as '%s'" % \
+        (dst_dir, args.output_format))
```

Comment:
```suggestion
    output_format = args.output_format or default_converter_name

    try:
        converter = env.converters[output_format]
    except KeyError:
        raise CliException("Converter for format '%s' is not found" %
            output_format)
    extra_args = converter.parse_cmdline(args.extra_args)

    dst_dir = args.dst_dir
    if dst_dir:
        if not args.overwrite and osp.isdir(dst_dir) and os.listdir(dst_dir):
            raise CliException("Directory '%s' already exists "
                "(pass --overwrite to overwrite)" % dst_dir)
    else:
        dst_dir = generate_next_file_name('%s-%s' % (
            make_file_name(args.dataset_id),
            make_file_name(output_format),
        ))
    dst_dir = osp.abspath(dst_dir)

    log.info("Downloading the dataset")
    extractor = extractor_factory()

    log.info("Exporting the dataset")
    converter.convert(extractor, dst_dir,
        default_image_ext='.png', **extra_args)

    log.info("Dataset exported to '%s' as '%s'" % (dst_dir, output_format))
```

`datum download -i tfds:cifar10 -- --save-images` fails because of `make_file_name(args.output_format)`.